cpp/src/io/parquet/experimental/page_index_filter.cu

```diff
@@ -627,6 +627,17 @@ std::unique_ptr<cudf::column> aggregate_reader_metadata::build_row_mask_with_pag
                                              static_cast<size_type>(output_dtypes.size())}
       .get_stats_columns_mask();
 
+  // Return early if no columns will participate in stats based page filtering
+  if (stats_columns_mask.empty()) {
+    auto row_mask = cudf::make_numeric_column(
+      data_type{cudf::type_id::BOOL8}, total_rows, rmm::device_buffer{}, 0, stream, mr);
+    thrust::fill(rmm::exec_policy_nosync(stream),
+                 row_mask->mutable_view().begin<bool>(),
+                 row_mask->mutable_view().end<bool>(),
+                 true);
```

Comment:
[`thrust::uninitialized_fill`](https://nvidia.github.io/cccl/thrust/api/group__filling_1ga309ec097df7d9064fbbd2f1f45ab3ab3.html) may be slightly faster